src/platform/Globals.cpp

```diff
@@ -25,14 +25,47 @@
 namespace chip {
 namespace DeviceLayer {
 
-chip::Inet::InetLayer InetLayer;
+chip::Inet::InetLayer & InetLayer()
+{
+    static chip::Inet::InetLayer gInetLayer;
+    return gInetLayer;
+}
 
-namespace Internal {
+#ifndef NDEBUG
```

Comment:
I think it needs to be possible to run tests in "release" builds. Testing is orthogonal to debugging.